### PR TITLE
Implement slow but accurate CloudFormation filtering

### PR DESCRIPTION
**Pull request by vektah**
The default filtering to exclude cloudformation resources when pulling is a simple match on the name to determine if it ends in an id. This can lead to a situation where both IAMY and CloudFormation are trying to manage the same resource when using explicitly named iam roles in cloudformation templates.

This PR implements a slower, more accurate way to determine if an item should be filtered out - it enumerates all CFN stacks and their resources, pops them into a map and checks it later.

Fixes https://github.com/99designs/iamy/issues/36

### CODE REVIEW

**Comment by lwc** [comment]
How much slower is it? should we just make this the default behaviour?

**Comment by vektah** [comment]
> How much slower is it? should we just make this the default behaviour?

Depends how many CFN stacks are in the account. For us it adds about a minute to what is normally a ~10 second pull.

I'm mostly just mindful of breaking any existing integrations out there.

File: iamy.go [hunk 1]
```diff
@@ -68,8 +69,9 @@ func main() {
 
 	case pull.FullCommand():
 		PullCommand(ui, PullCommandInput{
-			Dir:       *pullDir,
-			CanDelete: *canDelete,
+			Dir:                  *pullDir,
+			CanDelete:            *canDelete,
+			HeuristicCfnMatching: !*lookupCfn,
```
Comment (patrobinson) [inline iamy.go:74]: I'm finding it hard to figure out if this is on by default or not.
If you set this flag won't it disable it?

**Comment by patrobinson** [comment]
I haven't reviewed the code yet, but I'm skeptical about running this in an account with hundreds of stacks.
I have some I can test it against and will see what impact it has.

File: iamy.go (hunk 1 above)
Comment (vektah) [reply iamy.go:74]: passing `--accurate-cfn` will disable the faster heuristic matching.

**Comment by vektah** [comment]
> skeptical about running this in an account with hundreds of stacks.

Yeah, its likely not one size fits all so I haven't changed the default. Certainly being able to accurately exclude resources managed by cfn is important to us and we are happy to spend a little longer doing sync so that we dont:
 - cause drift in CFN stacks 
 - having a user managed in two places (cfn + iam users)

**Comment by dwurf** [comment]
For reference, real-world numbers on our dev account:
- with `--accurate-cfn --delete`: 187 stacks in 60 seconds
- Baseline (with `--delete` only): 187 stacks in 18 seconds

It's slower, but fast enough for me. I don't see how we can use iamy without this flag, given the amount of noise we are seeing

**Comment by simpsora** [comment]
A bit late to this party, but I'm curious what region is used for the CFN lookup.  I don't see any mention of region in the PR, so I imagine that the expectation is the caller has the region set via `AWS_REGION` or `AWS_DEFAULT_REGION`.

Region isn't meaningful for most IAM calls because of the service's global nature, but that's not true of CloudFormation.  If an IAM resource were defined in a CF stack in a different region than the current default, this change wouldn't pick it up.

Enumerating all CF stacks in all regions could be much more time-consuming than what's here already, but would be more accurate if one had resources in multiple regions.

**Pull request merged**